src/services/DateService.ts

```diff
@@ -388,6 +388,62 @@ export class DateService {
     return (end && start) ? moment.duration(end.valueOf() - start.valueOf()) : moment.duration(0);
   }
 
+  /**
+   * ## Returns formatted time:
+   * @param date the date reference (Date).
+   *
+   * ### Methods:
+   *
+   * #### `short`:
+   * Short formatted date string with the format of "MMM dd, yyyy"
+   *
+   * @param locale in which local to return the date. Defaults to "en-US"
+   * @returns Date
+   *
+   * #### `diff`:
+   * Return the time passed from <date> until now as the number of
+   * seconds, minutes, hours, days, weeks or years- depends on the
+   * difference size.
+   * * Up to 1 minutes, the output is "#sec".
+   * * Up to 1 hour, the output is "#min".
+   * * Up to 1 day, the output is "#h".
+   * * Up to 1 week, the output is "#d".
+   * * Up to 1 year, the output is "#w".
+   * * Over 1 year, the output is "#y".
+   * @returns string
+   *
+   * #### Example:
+   * formatDate(new Date()).short("de-DE");
+   *
+   * formatDate(new Date()).diff();
+   */
+  public formattedTime(date: Date | number): {short: (locale: string) => string, diff: () => string} {
+    const _date = new Date(date);
+
+    return {
+      short: (locale = "en-US") => _date.toLocaleDateString(
+        locale,
+        { year: "numeric", month: "short", day: "numeric" },
+      ),
+      diff: () => {
+        if (!date) return "-";
+        const diff = this.getDurationBetween(new Date(), _date);
+
+        if (diff.asMinutes() <= 1)
+          return diff.asSeconds().toFixed(0) + "sec";
+        if (diff.asHours() <= 1)
+          return diff.asMinutes().toFixed(0) + "min";
+        if (diff.asDays() <= 1)
+          return diff.asHours().toFixed(0) + "h";
+        if (diff.asWeeks() <= 1)
+          return diff.asDays().toFixed(0) + "d";
+        if (diff.asMonths() <= 12)
+          return diff.asWeeks().toFixed(0) + "w";
+
+        return diff.asMonths().toFixed(0) + "y";
+      },
+    };
+  }
```

Comment:
Do we need all these formats?  Can't we just use the existing `toString` by adding the formats we want to `formatArray`, and thus using built-in `moment` functionality instead of rolling our own?

Reply:
done using custom localization of moment.